optparse-applicative.cabal

```diff
@@ -51,10 +51,11 @@ source-repository head
   location: https://github.com/pcapriotti/optparse-applicative.git
 
 library
-  if impl(ghc >= 8)
-    ghc-options:       -Wall -Wno-redundant-constraints -Wcompat
-  else
-    ghc-options:       -Wall
+  ghc-options:         -Wall
+
+  -- See https://ghc.haskell.org/trac/ghc/wiki/Migration/8.0#base-4.9.0.0
+  if impl(ghc >= 8.0)
+    ghc-options:  -Wno-redundant-constraints -Wcompat -Wnoncanonical-monad-instances -Wnoncanonical-monadfail-instances
```

Comment:
`ghc-options` are combined via a concatenation monoid. That generally applies for all cabal  fields which are list-ish (such as options or `other-modules` or `build-depends` etc).